pkg/test/utils/kubernetes.go

```diff
@@ -952,23 +953,34 @@ func RunCommand(ctx context.Context, namespace string, command string, cmd harne
 		fmt.Sprintf("PATH=%s/bin/:%s", actualDir, os.Getenv("PATH")),
 	}
 
-	err = builtCmd.Run()
+	err = builtCmd.Start()
 	if err != nil {
 		if _, ok := err.(*exec.ExitError); ok && cmd.IgnoreFailure {
-			return nil
+			return nil, nil
 		}
+		return nil, err
 	}
 
-	return err
+	if cmd.Background {
+		return builtCmd, nil
+	}
+
+	err = builtCmd.Wait()
+	if _, ok := err.(*exec.ExitError); ok && cmd.IgnoreFailure {
+		return nil, nil
+	}
+	return nil, err
 }
 
 // RunCommands runs a set of commands, returning any errors.
 // If `command` is set, then `command` will be the command that is invoked (if a command specifies it already, it will not be prepended again).
-func RunCommands(logger Logger, namespace string, command string, commands []harness.Command, workdir string) []error {
+// commands running in the background are returned
```

Comment:
No, that isn't quite right. For commands running in foreground, `bgs` will have a `nil` entry. While this is a subtle difference, it can be a source of errors when mixing foreground and background in commands. Let's either be clear about this and have the caller distinguish between `nil` and non-`nil` entries or return `exec.Cmd` for all commands (as suggested for `RunCommand`) or only return `exec.Cmd` for background tasks.

Reply:
thank you!  made this statement correct... it is what was expect... a slice of only bg